.github/workflows/tests.yml

```diff
@@ -0,0 +1,80 @@
+name: tests
+
+on:
+  push:
+    paths-ignore:
+      - "*.md"
+      - "*.mdx"
+      - ".github/workflows/**"
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+    paths-ignore:
+      - "*.md"
+      - "*.mdx"
+
+jobs:
+  tests:
+    name: Test
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python_version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
```

Comment:
Perhaps we can look into limiting this grid more (e.g. testing 3.8, 3.9 and 3.10 on only one of the os's, like we do in spaCy) but for now it's not a big deal as the tests won't take up that much time, I suppose.

Reply:
Python 3.7 will reach EOL in two months anyway... https://devguide.python.org/versions/

Reply:
If we want to use `weasel` in spacy v3.6 then we need to test python 3.6+.

If we want to wait until spacy v4, then we could drop python 3.6-3.7.

Reply:
Yes, the current PR is fine. It's more to clarify that the supported python versions in weasel need to be based on the supported python versions for the relevant versions of spacy, not the python EOL dates.